### PR TITLE
Add pointed_thing_to_face_pos to misc_helpers and doc

### PR DESCRIPTION
**Pull request by MarkuBu**

### CODE REVIEW

**Comment by paramat** [comment]
:+1:

File: builtin/common/misc_helpers.lua [hunk 1]
```diff
@@ -671,3 +671,34 @@ else
...
+
+--------------------------------------------------------------------------------
+-- Returns the exact coordinate of a pointed surface
+--------------------------------------------------------------------------------
+function core.pointed_thing_to_face_pos(placer, pointed_thing)
+	local eye_offset_first = placer:get_eye_offset()
+	local node_pos = pointed_thing.under
+	local camera_pos = placer:get_pos()
+	local pos_off = vector.multiply(
+			vector.subtract(pointed_thing.above, pointed_thing.under), 0.5)
```
Comment (rubenwardy) [inline builtin/common/misc_helpers.lua]: vector.subtract(pointed_thing.above, node_pos), 0.5)

File: doc/client_lua_api.md [hunk 2]
```diff
@@ -787,6 +787,8 @@ Call these functions only at load time!
       extra arguments and return the result
 * `fgettext(string, ...)` : returns string
     * same as fgettext_ne(), but calls core.formspec_escape before returning result
+* `minetest.pointed_thing_to_face_pos(placer, pointed_thing)`: returns a table
```
Comment (rubenwardy) [inline doc/client_lua_api.md]: returns a position

Comment (paramat) [inline doc/client_lua_api.md]: Yeah 'table' seemed odd to me too.

**Comment by MarkuBu** [comment]
> this could use some comments, or some less short variable names, as it's fairly unreadable

Didn't write the original code, so I don't know myself what some variables mean

@HybridDog

File: doc/client_lua_api.md (hunk 2 above)
Comment (MarkuBu) [inline doc/client_lua_api.md]: I think "table" is correct. What's the content of the table is described in the next line. And the same term is used for other functions too

Just two examples from client_lua_api.md
```
* `minetest.deserialize(string)`: returns a table
* `minetest.serialize(table)`: returns a string
```

**Comment by HybridDog** [comment]
l added comments there: https://github.com/minetest/minetest/compare/master...HybridDog:fine_pointed#diff-251d22e777d517243ee7aa4c6b4aa60eR251

pos_off: the vector from node_pos to the face
offset: the distance with sign from node_pos to the face
nc: normal coordinate; the coordinate where pos_off is not 0
oc: other coordinates; the two other coordinates of pos_off
c: coordinate
v: value
f: factor

File: doc/client_lua_api.md (hunk 2 above)
Comment (kaeza) [inline doc/client_lua_api.md]: @MarkuBu `deserialize` returns a table for which we don't know the keys beforehand, while this new function does, so your example is irrelevant. I think @rubenwardy's comment is correct.

**Comment by kaeza** [comment]
Could this be generalized somehow to work with two arbitrary positions?

**Comment by MarkuBu** [comment]
> nc: normal coordinate; the coordinate where pos_off is not 0
oc: other coordinates; the two other coordinates of pos_off
c: coordinate

I think coordinate is the wrong term. I need a while to understand what's wrong. A better term would be "axis"

> Could this be generalized somehow to work with two arbitrary positions?

Not by me. I want that this get's merged or taken by someone else because I want to delete my MTG fork. I broke something in the master branch and I don't know how to repair it

**Comment by paramat** [comment]
Yes we can improve this later.

File: doc/client_lua_api.md (hunk 2 above)
Comment (paramat) [inline doc/client_lua_api.md]: We don't refer to a pos as a table, even though it is. 'Table' suggests a table of multiple positions and is confusing.

**Comment by paramat** [comment]
:+1: Once the 2 comments are addressed.

**Comment by HybridDog** [comment]
> A better term would be "axis"

You're right, that was one of the reasons l used short variables.

File: doc/client_lua_api.md (hunk 2 above)
Comment (MarkuBu) [inline doc/client_lua_api.md]: agreed

**Comment by paramat** [comment]
Thannks :+1:

**Comment by rubenwardy** [comment]
> Yes we can improve this later.

this never happens :P

**Comment by paramat** [comment]
Often yes, but:

> Not by me. I want that this get's merged or taken by someone else because I want to delete my MTG fork. I broke something in the master branch and I don't know how to repair it

**Comment by MarkuBu** [comment]
I'll keep this in mind, cut currently this PR blocks me

**Comment by paramat** [comment]
I don't expect you to make a follow up PR though, i meant improvement PRs can come later and this can probably be merged now.

File: doc/client_lua_api.md (hunk 2 above)
Comment (DS-Minetest) [inline doc/client_lua_api.md]: Suggestion: Write something like `returns a vector`.

**Comment by paramat** [comment]
I can guess you will not want to rebase this?
Give me a few hours to grab the diff then i'll open a fresh PR.

**Comment by paramat** [comment]
Now copied the diff so you can close this if you need to.

**Comment by HybridDog** [comment]
paramat, the patch is still available after closing or deleting the repository. It only disappears if the branch is reset to the master branch and then force pushed.
```$ wget https://patch-diff.githubusercontent.com/raw/minetest/minetest/pull/5439.patch```

**Comment by paramat** [comment]
@MarkuBu please could you let me know what you will do? if you are happy to rebase this that's great but if you are not going to rebase this let me know and close this so i can work on the fresh PR.

**Comment by MarkuBu** [comment]
I'll make a new PR. Still don't know how to rebase

**Comment by paramat** [comment]
Thanks, no problem about rebasing, it can be complex.

**Comment by nerzhul** [comment]
Don't forget to close this

**Comment by MarkuBu** [comment]
I try to rebase first. I want to learn something. I it doesn't work I make a new PR

**Comment by MarkuBu** [comment]
Replaced by https://github.com/minetest/minetest/pull/5485

**Pull request closed without merging**